### ISSUE TITLE
🐛 Regex field in dictionary is not implemented

### ISSUE THREAD

**Issue by hknahal** [icgc-argo/argo-dictionary#405]
## Describe the bug
During the `Register Samples` step, if a data submitter submits a value in the `submitter_donor_id`, `submitter_specimen_id` or  `submitter_sample_id` fields which includes a comma character, it validates. However, the regular expression in the dictionary does not allow a comma, so it's unclear how values with a comma are valid in the submission system.

## Steps To Reproduce
1. Go to LUCA-KR Program Dashboard (https://platform.icgc-argo.org/submission/program/LUCA-KR/dashboard) and download `ALL CLINICAL DATA`
2. Open `sample_registration.tsv` file
3. Find to `submitter_donor_id` "34991616":
The programs appears to have submitted a submitter_specimen_id with a comma character:

![multiple_specimens](https://user-images.githubusercontent.com/4992044/235756403-212c00c7-4b29-4937-b56b-38eb9959f139.png)

4. But the regex in the dictionary doesn't allow for a comma in the submitted value:
```
{
  "submitter_id": "^[A-Za-z0-9\\-\\._]{1,64}"
}
```

This suggests that perhaps the `regex` field in the dictionary is not being implemented by clinical:

![submitter_specimen_id](https://user-images.githubusercontent.com/4992044/235759147-0e58818f-03a3-4d74-8590-97fd116b2695.png)

## Expected behaviour
The validation should fail if it does not match the regular expression in the dictionary.

**Comment by hknahal** [comment]
Another example suggesting the `regex` field in the dictionary isn't being implemented:

LUCA-KR program was able to submit a value consisting of bracket characters in the `submitter_specimen_id` field in the sample_registration table:

![submitter_specimen_id_with_brackets](https://user-images.githubusercontent.com/4992044/235761114-d7f6ccec-942b-4b3a-ba4a-e09b7d790df3.png)


But the regex in the dictionary shouldn't allow this value to be valid:
```
{
  "submitter_id": "^[A-Za-z0-9\\-\\._]{1,64}"
}
```

**Comment by hknahal** [comment]
Another example of where it appears the "regex" field is not implemented in Clinical:

- `anatomic_site_progression_or_recurrence` has a regex property:
![regex](https://user-images.githubusercontent.com/4992044/236325880-b8a55f3f-3ce8-424b-b403-6315a05d4ebd.png)

When testing clinical submissions in QA using dictionary version 125.2, the following values for `anatomic_site_progression_or_recurrence` should have failed validation because they do not match the regex:

D23.1 --> Regex only allows codes that begin with the letter "C"
blahblah  -> Regex should not allow this, as it expects an ICD-O-3 formatted code.

![follow_up](https://user-images.githubusercontent.com/4992044/236326914-ecc8a72e-1ff8-4d69-8bd3-ccdb5f992702.png)

**Comment by hknahal** [comment]
Testing in Prod using `TEST-CA`:

1. Upload [sample_registration.txt](https://github.com/icgc-argo/argo-dictionary/files/12110504/sample_registration.txt) file in `Register Samples` in `TEST-CA` program.
2. Upon upload, we should have expected an error message because the `submitter_specimen_id` has a comma which the dictionary regex does not allow, but it passed validation:
![luca-kr](https://github.com/icgc-argo/argo-dictionary/assets/4992044/7547c325-1e4c-4df1-9a1b-93747a7e06dc)

**Comment by Buwujiu** [comment]
questions: 1. is this happening to other submitter ids? 2. It's also happening in QA.

**Comment by hknahal** [comment]
Confirmed this behavior happens for all submitter IDs in both Prod and QA. 

Sample submission files uploaded to `TEST-CA`:
[sample_registration_luca-kr_incorrect_ids_submitted_to_test-ca.txt](https://github.com/icgc-argo/argo-dictionary/files/12443246/sample_registration_luca-kr_incorrect_ids_submitted_to_test-ca.txt)

[donor_v1.18.txt](https://github.com/icgc-argo/argo-dictionary/files/12443243/donor_v1.18.txt)
[specimen_v1.18.txt](https://github.com/icgc-argo/argo-dictionary/files/12443253/specimen_v1.18.txt)
[primary_diagnosis_v1.18.txt](https://github.com/icgc-argo/argo-dictionary/files/12443245/primary_diagnosis_v1.18.txt)
[treatment_v1.18.txt](https://github.com/icgc-argo/argo-dictionary/files/12443247/treatment_v1.18.txt)
[chemotherapy_v1.18.txt](https://github.com/icgc-argo/argo-dictionary/files/12443242/chemotherapy_v1.18.txt)
[follow_up_v1.18.txt](https://github.com/icgc-argo/argo-dictionary/files/12443244/follow_up_v1.18.txt)

Examples:
Able to register incorrect LUCA-KR identifiers with comma, brackets and tilde characters:
![luca-kr_incorrect_ids_registered](https://github.com/icgc-argo/argo-dictionary/assets/4992044/07e85ff6-42f7-4771-9520-cd08b35e59a1)

Was able to validate incorrect characters in follow up file:
![follow-up_incorrect_char](https://github.com/icgc-argo/argo-dictionary/assets/4992044/f157223a-0e4c-4065-9b6b-ce73acc21dac)

**Comment by demariadaniel** [comment]
https://github.com/icgc-argo/argo-dictionary/pull/412

**Comment by hknahal** [comment]
Can close this ticket since it was released in Dictionary 1.19
